tests/datasource/fluent/integration/test_integration_datasource.py

```diff
@@ -525,19 +541,124 @@ def test_spark_data_adding_dataframe_in_file_reloaded_context(
     context = empty_file_context
 
     dataframe_asset: SparkDataFrameAsset = context.data_sources.add_or_update_spark(
-        name="fluent_pandas_datasource"
+        name="fluent_spark_datasource"
     ).add_dataframe_asset(name="my_df_asset")
-    _ = dataframe_asset.build_batch_request(dataframe=spark_df)
-    assert dataframe_asset.dataframe.toPandas().equals(df)  # type: ignore[union-attr]
-
-    datasource = context.data_sources.add_or_update_spark(name="fluent_pandas_datasource")
-    dataframe_asset = datasource.add_dataframe_asset(name="my_df_asset")
-    _ = dataframe_asset.build_batch_request(dataframe=spark_df)
-    assert dataframe_asset.dataframe.toPandas().equals(df)  # type: ignore[union-attr]
+    batch_def = dataframe_asset.add_batch_definition_whole_dataframe(name="bd")
+    batch = batch_def.get_batch(batch_parameters={"dataframe": spark_df})
+    assert isinstance(batch.data, SparkDFBatchData)
+    assert batch.data.dataframe.toPandas().equals(df)
 
     context = gx.get_context(context_root_dir=context.root_directory, cloud_mode=False)
-    dataframe_asset = context.get_datasource(datasource_name="fluent_pandas_datasource").get_asset(
-        asset_name="my_df_asset"
+    retrieved_bd = (
+        context.get_datasource(datasource_name="fluent_spark_datasource")
+        .get_asset(asset_name="my_df_asset")
+        .get_batch_definition(batch_definition_name="bd")
+    )
+    new_batch = retrieved_bd.get_batch(batch_parameters={"dataframe": spark_df})
+    assert isinstance(new_batch.data, SparkDFBatchData)
+    assert new_batch.data.dataframe.toPandas().equals(df)
+
+
+@dataclass
```

Comment:
From here to the end of the file are new e2e tests that verify the validation workflows for batches and validation_defintions.